src/main/java/domain/Car.java

```diff
@@ -0,0 +1,64 @@
+package domain;
+
+import static domain.ExceptionMessages.EMPTY_CAR_NAME_EXCEPTION;
+import static domain.ExceptionMessages.SPECIAL_CHARACTER_CAR_NAME_EXCEPTION;
+import static domain.ExceptionMessages.TOO_LONG_CAR_NAME_EXCEPTION;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+public class Car {
+    private static final int MAX_NAME_LENGTH = 5;
+    private static final int MOVE_FORWARD_LOWER_BOUND = 3;
+    private final String name;
+    private int distance = 0;
+
+    public Car(final String name) {
+        validateCarName(name);
+        this.name = name;
+    }
+
+    private void validateCarName(String name) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException(EMPTY_CAR_NAME_EXCEPTION.getMessage());
+        }
+        if (name.length() > MAX_NAME_LENGTH) {
+            throw new IllegalArgumentException(TOO_LONG_CAR_NAME_EXCEPTION.getMessage());
+        }
+        Pattern pattern = Pattern.compile("[ !@#$%^&*().?\":{}|<>]");
```

Comment:
정규표현식은 무엇을 의미하는지 한 눈에 들어오지 않는 경우가 많아서 이름을 지어주면 더 좋을거 같습니다.
```suggestion
        Pattern findSpecialCharacterPattern = Pattern.compile("[ !@#$%^&*().?\":{}|<>]");
```

또 패턴 컴파일은 비용이 큰 작업이어서, pattern을 static으로 미리 load해두고 재활용하는걸 추천드립니다.

Reply:
정규표현식 사용이 익숙하지 않다보니 어떻게 쓰는게 가장 효율적일지 미처 생각하지 못했습니다!
좋은 지적 감사합니다!